gpio_test.go

```diff
@@ -0,0 +1,173 @@
+package gpio
+
+import (
+	"os"
+	"strconv"
+	"strings"
+	"testing"
+	"time"
+
+	"github.com/stretchr/testify/assert"
+)
+
+// validate and extract test setup
+func getEnv(t *testing.T) (gpioDev string, gpioIn uint32, gpioOut uint32, hasGpio bool, hasLoopback bool) {
```

Comment:
Now that there are so many return values, maybe struct is better. Your choice.

Examples:
- https://github.com/temoto/vender/blob/062e69b44c6a9bd1eeb613a86ae75227cefd5887/head/tele/tele_test.go
- https://github.com/temoto/vender/blob/0a1110c8ce7d0f713d29dfbda1a75b01f0fbf9ba/head/ui/util_test.go#L18